src/ol/array.js

```diff
@@ -1,8 +1,65 @@
 goog.provide('ol.array');
 
-goog.require('goog.array');
 goog.require('goog.asserts');
 
+function defaultCompare(a, b) {
+  return a > b ? 1 : a < b ? -1 : 0;
+}
+
+/**
+ * Performs a binary search on the provided sorted list and returns the index of the item if found. If it can't be found it'll return -1.
+ * https://github.com/darkskyapp/binary-search
+ *
+ * @param {Array<*>} haystack Items to search through.
+ * @param {*} needle The item to look for.
+ * @param {Function=} opt_comparator Comparator function
+ * @param {number=} low Lower bounds.
+ * @param {number=} high Higher bounds.
+ * @return {number} The index of the item if found, -1 if not.
+ */
+ol.array.binarySearch = function(haystack, needle, opt_comparator, low, high) {
```

Comment:
You can also keep the binarySearch function you always have, but please remove the low and high arguments. Then you can also remove the checks and `throw`s for invalid bounds, resulting in a simpler and faster function too.